BigQuery/src/Model.php

```diff
@@ -0,0 +1,229 @@
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/get Models get API documentation.
+     *
+     * @param array $options
+     * @return array
+     */
+    public function reload(array $options = [])
+    {
+        return $this->info = $this->connection->getModel($options + $this->identity);
+    }
+
+    /**
+     * Retrieves the model's ID.
+     *
+     * Example:
+     * ```
+     *  echo $model->id();
+     * ```
+     *
+     * @return string
+     */
+    public function id()
+    {
+        return $this->identity['modelId'];
+    }
+
+    /**
+     * Retrieves the model's identity.
+     *
+     * An identity provides a description of a resource that is nested in nature.
+     *
+     * Example:
+     * ```
+     * echo $model->identity()['modelId'];
+     * ```
+     * @return array
+     */
+    public function identity()
+    {
+        return $this->identity;
+    }
+
+    /**
+     * Delete the model.
+     *
+     * Please note that by default the library will not attempt to retry this
+     * call on your behalf.
+     *
+     * Example:
+     * ```
+     * $model->delete();
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/delete Models delete API documentation.
+     *
+     * @param array $options [optional]
+     */
+    public function delete(array $options = [])
+    {
+        $this->connection->deleteModel(
+            $options
+            + ['retries' => 0]
+            + $this->identity
+        );
+    }
+
+    /**
+     * Check whether or not the model exists
+     *
+     * Example:
+     * ```
+     * echo $model->exists();
+     * ```
+     * @return bool
+     */
+    public function exists()
+    {
+        try {
+            $this->connection->getModel($this->identity);
+        } catch (NotFoundException $ex) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Update the model.
+     *
+     * Providing an `etag` key as part of `$metadata` will enable simultaneous
+     * update protection. This is useful in preventing override of modifications
+     * made by another user. The resource's current etag can be obtained via a
+     * GET request on the resource.
+     *
+     * Please note that by default this call will not automatically retry on
+     * your behalf unless an `etag` is set.
+     *
+     * Example:
+     * ```
+     * $model->update([
+     *     'friendlyName' => 'My ML model'
+     * ]);
+     * ```
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/patch Model's patch API documentation.
+     * @see https://cloud.google.com/bigquery/docs/api-performance#patch Patch (Partial Update)
+     *
+     * @param array $metadata
+     * @param array $options
+     * @return array
+     */
+    public function update(array $metadata, array $options = [])
+    {
+        $options = $this->applyEtagHeader(
+            $options
```

Comment:
nit: `$options` should be the last array in a list of merged arrays. That'll prevent the user from overriding something you set specifically in the client code.

Reply:
Makes sense to me but Dataset has it written this way!

Reply:
Making the change, maybe we should think about making the change in Dataset in the future too.

Reply:
Indeed, that's just old code. I try to fix it when I touch the methods where it's in a less than optimal order.